vignettes/intro.Rmd

```diff
@@ -0,0 +1,165 @@
+# To display a scatterplot
+ggplot(mpg) +
+  aes(x = displ, y = hwy) +
+  geom_point()
+
+# To fit a loess trendline to the data
+ggplot(mpg) +
+  aes(x = displ, y = hwy) +
+  stat_smooth(formula = y ~ x, method = "loess")
+```
+
+## Scales
+
+[Scales](https://ggplot2-book.org/scales-guides.html) are important for translating what is shown on the graph back to an understanding of the data. The scales typically form pairs with aesthetic attributes of the plots, and are represented in plots by guides, like axes or legends.
+Scales are responsible for updating the limits of a plot, setting the breaks, formatting the labels and possibly applying a transformation.
+
+To use scales, one can use one of the scale functions that are patterned as `scale_{aesthetic}_{type}()` functions, where `{aesthetic}` is one of the pairings made in the mapping part of a plot. To map the `class` column in the `mpg` dataset to the viridis colour palette, we can write the following:
+
+```{r example_scales}
+ggplot(mpg) +
+  aes(x = displ, y = hwy, colour = class) +
+  geom_point() +
+  scale_colour_viridis_d()
+```
+
+## Facets
+
+[Facets](https://ggplot2-book.org/facet.html) can be used to separate small multiples, or different subsets of the data. 
+It is a powerful tool to quickly split up the data into smaller panels to look for meaning among the subsets.
+In addition, the layout can convey a pattern because more than one variable may be used to split the data.
+
+The facets have their own mapping that can be given to the `vars()` function. 
+To show the `mpg` dataset by the `drv` and `year` columns, we can use `facet_grid()` as follows:
+
+```{r example_facets}
+ggplot(mpg) +
+  aes(x = displ, y = hwy) +
+  geom_point() +
+  facet_grid(rows = vars(year), cols = vars(drv))
+```
+
+## Coordinates
+
+You can view the [coordinates](https://ggplot2-book.org/coord.html) part of the plot as an interpreter of positional aesthetics. 
+While typically Cartesian coordinates are used, the coordinate system powers the display of [map](https://ggplot2-book.org/maps.html) projections and [polar](https://ggplot2-book.org/coord.html#polar-coordinates-with-coord_polar) plots.
+
+To show the `mpg` dataset in polar coordinates instead of Cartesian ones, one can use `coord_polar()` to determine what should be mapped to the angle (theta) or radius.
+
+```{r example_coords}
+ggplot(mpg) +
+  aes(x = displ, y = hwy) +
+  geom_point() +
+  coord_polar(theta = "y")
+```
+
+## Theme
+
+The [theme](https://ggplot2-book.org/polishing.html) system controls almost any visuals of the plot that are not controlled by the data, and is therefor important for the look and feel of the plot. You can use the theme for things ranging from determining the location of the legends to setting the horizontal text adjustment for the bottom x axis specifically. Many elements in the theme are hierarchical in that setting the look of the general axis line affects those of the bottom and left axes simultaneously. 
```

Comment:
```suggestion
The [theme](https://ggplot2-book.org/polishing.html) system controls almost any visuals of the plot that are not controlled by the data and is therefore important for the look and feel of the plot. You can use the theme for things ranging from determining the location of the legends to setting the horizontal text adjustment for the bottom x-axis specifically. Many elements in the theme are hierarchical in that setting the look of the general axis line affects those of the bottom and left axes simultaneously.
```

Reply:
> You can use the theme for things ranging from determining the location of the legends to setting the horizontal text adjustment for the bottom x axis specifically.

"things" is vague in this sentence, it's better/more informative to use actual terms than placeholders.

Reply:
Agreed, fair point.